File: Algorithms/Sorting_Algorithms/Selection_Sort/Selection_Sort.cpp
//Selection sort, sort the array by swapping the curr element with the smallest element from indexes curr+1 to len-1
#include<iostream>
using namespace std;
void selectionsort(int arr[],int len){
    for(int curr=0;curr<len;curr++){
        int min_idx=curr;
        for(int itr=curr+1;itr<len;itr++){
            if(arr[itr]<arr[min_idx]){
                min_idx=itr;
            }
        }
        swap(arr[curr],arr[min_idx]);
    }
}
int main(){
    int len;
    cout<<"Enter the number of elements to sort: ";
    cin>>len;
    int arr[len]={0};
    cout<<"Enter the elements to sort: \n";
    for(int idx=0;idx<len;idx++){
        cin>>arr[idx];
    }
    selectionsort(arr,len);
    cout<<"Sorted array: ";
    for(int idx=0;idx<len;idx++){
        cout<<arr[idx]<<" ";
    }
    return 0;
}

File: Algorithms/Sorting_Algorithms/Selection_Sort/SelectionSort.py
# This is the Python version of Selection Sort
# Code is contributed by: Italo Vinicius.
# 
# Selection sort is an Sorting Algorithm which takes compelxity: O(n²)

# Function to do selection sort 
def selectionSort(arr): 
  
    # Traverse through all array elements 
    for i in range(len(arr)): 
        
        # Find the minimum element in remaining  
        # unsorted array 
        min_idx = i 
        for j in range(i+1, len(arr)): 
            if arr[min_idx] > arr[j]: 
                min_idx = j 
                
        # Swap the found minimum element with  
        # the first element         
        arr[i], arr[min_idx] = arr[min_idx], arr[i]  

if __name__ == "__main__":
    arr = list(map(int,input().split()))
    selectionSort(arr)
    
    print(' '.join(str(x) for x in arr))